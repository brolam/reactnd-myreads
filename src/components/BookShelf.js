import React from 'react';
import PropTypes from 'prop-types'
import Book from './Book.js';

function BookShelf(props) {
    const { shelfTitle, books, onChangeBookShelf } = props
    return (
        <div className="bookshelf">
            <h2 className="bookshelf-title">{shelfTitle}</h2>
            <div className="bookshelf-books">
                <ol className="books-grid">
                    {books.map(book => {
                        return (
                            <Book
                                key={book.id}
                                book={book}
                                onChangeBookShelf={onChangeBookShelf}
                            />
                        )
                    })}
                </ol>
            </div>
        </div>
    );
}

BookShelf.propTypes = {
    shelfTitle: PropTypes.string.isRequired,
    books: PropTypes.array.isRequired,
    onChangeBookShelf: PropTypes.func.isRequired,
}

export default BookShelf