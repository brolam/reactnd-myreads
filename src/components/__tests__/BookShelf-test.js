import React from 'react';
import BookShelf from '../BookShelf.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

const books = [{
    id: "nggnmAEACAAJ",
    title: "The Linux Command Line",
    authors: ["William E. Shotts", "Breno Marques"],
    shelf: "currentlyReading"},
    {
    id: "PGR2AwAAQBAJ",
    title: "To Kill a Mockingbird",
    authors: ["Harper Lee", "Breno Marques"],
    shelf: "currentlyReading"}
]

const onChangeBookShelf = () => { }

test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BookShelf shelfTitle="Currently Readin" books={books} onChangeBookShelf={onChangeBookShelf} />, div)
})

test('renders 2 books', () => {
    const bookShelf = mount(<BookShelf shelfTitle="Currently Readin" books={books} onChangeBookShelf={onChangeBookShelf} />);
    const booksOnTheShelf = bookShelf.find('.book');
    expect(booksOnTheShelf.length).toEqual(2);
  })

test('Last Snapshot', () => {
    const bookShelf = renderer.create(
        <BookShelf shelfTitle="Currently Readin" books={books} onChangeBookShelf={onChangeBookShelf} />
    );
    let tree = bookShelf.toJSON();
    expect(tree).toMatchSnapshot();
})