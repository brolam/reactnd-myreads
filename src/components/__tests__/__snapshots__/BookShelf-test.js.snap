// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Snapshot 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Currently Readin
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(\\"https://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1\\")",
                  "height": 190,
                  "width": 120,
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select
                id="nggnmAEACAAJ"
                onChange={[Function]}
                value="currentlyReading"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            The Linux Command Line
          </div>
          <div
            className="book-authors"
          >
            William E. Shotts, 
            Breno Marques 
          </div>
        </div>
      </li>
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(\\"https://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1\\")",
                  "height": 190,
                  "width": 120,
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select
                id="PGR2AwAAQBAJ"
                onChange={[Function]}
                value="currentlyReading"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            To Kill a Mockingbird
          </div>
          <div
            className="book-authors"
          >
            Harper Lee, 
            Breno Marques 
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
